### ISSUE TITLE
Sketchup::Importer superclass has undocumented #show_summary() callback

### ISSUE THREAD

**Issue by DanRathbun** [SketchUp/api-issue-tracker#77]
#### [`Sketchup::Importer`](http://ruby.sketchup.com/Sketchup/Importer.html) 

[`Sketchup::Importer`](http://ruby.sketchup.com/Sketchup/Importer.html) superclass has undocumented `#show_summary()` callback.

Does the SketchUp importer interface poll Ruby importer objects for this callback ?
* If so, docs need update.
* If not, C implementation should remove this unused proto-method.

----

Related. Why are there even proto methods that all return **`false`** ?
Ie, why is this class not abstract like the observers ?

**Comment by thomthom** [comment]
Logged as SU-40206

**Comment by Eneroth3** [comment]
@thomthom What is the intention here? Should show_summary be documented for public consumption or is it for internal use?

**Comment by thomthom** [comment]
I found this in the source:

```
Note that I could not figure out how to get this
// method to fire, so I am not including API documentation for it at this
// point. We should determine if this is ever called and either remove it
// or add correct documentation. 2-12-09
```

So it's intentionally not documented.

**Comment by DanRathbun** [comment]
As said in 1st report ...

> If not, C implementation should remove this unused proto-method.

All observers are now abstract, and all OTHER false returning or empty proto-callback methods were removed from the observer superclasses, ... **except this one**.

So, please reopon the issue until it is removed from the class definition on the C-side.

**Comment by thomthom** [comment]
Does this have any impact on extension development at all?

**Comment by DanRathbun** [comment]
It might, ... I don't know. It's a cleanup issue. If I noticed it, ... others may also. And they may ask as I did. (Which may mean you'll continue to answer this question periodically.

You (the team) have released a document stating that all observers are abstract and that you removed all "proto-methods".
I have shown that this is not entirely true by logging this issue as a **documentation issue.**

I can **guess** that it was supposed to be called if a certain code was returned from the [`Sketchup::Importer#load_file`](https://ruby.sketchup.com/Sketchup/Importer.html#load_file-instance_method) callback, ... **but** at the last minute someone just decided that (it might be best if) the coder could handle the display of import results better within that callback's logic, rather than figure out a way to get text from the `load_file()` callback to a `show_summary()` callback. (Which would compete with the need to get an error code returned.)

Ie ... an importer subclass would make the call to the **custom** local `show_summary()` method from within the `load_file()` logic in a manner similar to ...
```ruby
def show_summary(summary_text, file_path)
  UI.messagebox(summary_text, MB_MULTILINE, "Import Summary : #{File.basename(file_path)}")
end
```

Of course they would be free to use a nicer HTML based dialog if desired.
